src/Microsoft.Extensions.Caching.StackExchangeRedis/RedisCacheServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.StackExchangeRedis;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for setting up Redis distributed cache related services in an <see cref="IServiceCollection" />.
+    /// </summary>
+    public static class RedisCacheServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds Redis distributed caching services to the specified <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="setupAction">An <see cref="Action{RedisCacheOptions}"/> to configure the provided
+        /// <see cref="RedisCacheOptions"/>.</param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        public static IServiceCollection AddDistributedStackExchangeRedisCache(this IServiceCollection services, Action<RedisCacheOptions> setupAction)
```

Comment:
AddStackExchangeRedisCache?

Reply:
Who would make the decision here?

Reply:
@Eilon or @DamianEdwards

Reply:
Per discussion, we can drop the `Distributed` from here, though I'm fine as-is, or even `AddStackExchangeRedisDistributedCache`